sdk/storage/Azure.Storage.DataMovement/tests/TransferManagerTests.cs

```diff
@@ -255,20 +281,35 @@ public async Task BasicContainerTransfer(
             srcResource.VerifyNoOtherCalls();
             dstResource.VerifyNoOtherCalls();
         }
+        foreach (MemoryTransferCheckpointer.Job job in checkpointer.Jobs.Values)
+        {
+            Assert.That(job.Parts.Count, Is.GreaterThan(1), "Containers should have several parts.");
```

Comment:
A little nitpicky here but I think you can assert the exact number of job parts using GetItemCountFromContainerIndex()

Reply:
I can't guarantee dictionary ordering. I can more easily guarantee that the expected number of parts made it into the checkpointer in some form for this test.